### PR TITLE
chore: added sql-validator in git-hub action

### PR DESCRIPTION
**Pull request by badal773**
This action is designed to check whether the SQL files located in the '/scripts/sql' directory are in the correct format, identify if there are any repeated files.

### CODE REVIEW

**Review by tayalrishabh96 — DISMISSED** [review]

File: .github/workflows/pr-issue-validator.yaml [hunk 1]
```diff
@@ -120,3 +124,59 @@ jobs:
             gh pr edit $PRNUM --remove-label "PR:Ready-to-Review"
             exit 1
         fi
+    - name: Check SQL file format and duplicates
+      shell: bash
+      env: 
+          pr_no: ${{ github.event.pull_request.number }}
+          GH_TOKEN: ${{ github.token }}
+      run: |
+          echo "GitHub Token: $GH_TOKEN"
```
Comment (tayalrishabh96) [inline .github/workflows/pr-issue-validator.yaml:133]: remove this.

File: .github/workflows/pr-issue-validator.yaml [hunk 2]
```diff
@@ -120,3 +124,59 @@ jobs:
...
+            fi
+            echo "Current directory: $(pwd)"
+            
+            cd scripts/sql/
+            echo "SQL files directory: $(pwd)"
+            
+            migration_no=$(echo $filename | cut -d "/" -f 3 | cut -d "_" -f 1)
+            echo "Migration Number: $migration_no"
+            
+            migration_files_of_this_no=$(ls | cut -d "_" -f 1 | grep  -w -c  $migration_no )
```
[the remark this answers is not in the document]
Comment (tayalrishabh96) [reply .github/workflows/pr-issue-validator.yaml:163]: Also add comment for this line, what is it doing ?

File: .github/workflows/pr-issue-validator.yaml [hunk 3]
```diff
@@ -120,3 +124,59 @@ jobs:
...
+            cd ../../
+            echo "Current directory: $(pwd)"
+            
+            echo "Migration Number: $migration_no"
+            echo "Number of files with this migration number: $migration_files_of_this_no"
+            
+            if [[ $migration_files_of_this_no == "2" ]] ; then
+              echo "All looks good for this migration number."
+            elif [[ $migration_files_of_this_no == "1" ]] ; then
+              echo "Not Looks good . only $filename this file present."
```
Comment (tayalrishabh96) [inline .github/workflows/pr-issue-validator.yaml:173]:
```suggestion
              echo "Only single migration file was present for migration no.: $migration_no. either up or down migration is missing! EXITING"
```

File: .github/workflows/pr-issue-validator.yaml [hunk 4]
```diff
@@ -120,3 +124,59 @@ jobs:
...
+              echo "Only single migration file was present for this No-  $migration_no"
+              gh pr comment $pr_no --body "Error: Only Single migration file was present for this no -: $migration_no."
+              exit 1
+            else 
+               echo "Error: Migration number is repeated."
+               gh pr comment $pr_no --body "Error: The SQL file number is duplicated: $migration_no."
```
Comment (tayalrishabh96) [inline .github/workflows/pr-issue-validator.yaml:179]:
```suggestion
               gh pr comment $pr_no --body "Error: The SQL file number: $migration_no is duplicated"
```

File: .github/workflows/pr-issue-validator.yaml (hunk 1 above)
Comment (tayalrishabh96) [inline .github/workflows/pr-issue-validator.yaml:133]:
```suggestion
```

File: .github/workflows/pr-issue-validator.yaml [hunk 5]
```diff
@@ -120,3 +124,59 @@ jobs:
...
+          echo "GitHub Token: $GH_TOKEN"
+          git fetch origin main
+          git diff origin/main...HEAD --name-only > mydiff
+          echo "Changed SQL files:"
+          cat mydiff
+          file_list=$(git diff origin/main...HEAD --name-only)
+          awk '/scripts\/sql\//' mydiff
+          count=$(awk '/scripts\/sql\//' mydiff | wc -l) 
+          if [[ $count == "0" ]] ; then 
+            echo "No Sql file is added .exiting from this action"
```
Comment (tayalrishabh96) [inline .github/workflows/pr-issue-validator.yaml:142]:
```suggestion
            echo "No Sql file is added, exiting from this github action"
```

File: .github/workflows/pr-issue-validator.yaml [hunk 6]
```diff
@@ -120,3 +124,59 @@ jobs:
...
+            exit 0
+          fi
+          for filename in $file_list; do
+            echo "Checking File: $filename"
+            
+            if [[ "$filename" =~ \.(up|down)\.sql$ ]]; then
+              echo "File name is in the correct format: $filename"
```
Comment (tayalrishabh96) [inline .github/workflows/pr-issue-validator.yaml:149]:
```suggestion
              echo "File name: $filename is in the correct format"
```

File: .github/workflows/pr-issue-validator.yaml [hunk 7]
```diff
@@ -120,3 +124,59 @@ jobs:
...
+            else
+              echo "Error: File name is not in the correct format: $filename."
+              gh pr comment $pr_no --body "Error: The SQL file name is not in the correct format: $filename."
```
Comment (tayalrishabh96) [inline .github/workflows/pr-issue-validator.yaml:152]:
```suggestion
              gh pr comment $pr_no --body "Error: The SQL file name: $filename is not in the correct format"
```

File: .github/workflows/pr-issue-validator.yaml (hunk 4 above)
Comment (badal773) [reply .github/workflows/pr-issue-validator.yaml:179]: DONE

**Review by tayalrishabh96 — COMMENTED** [review]
looks good, can be merged.

**Review by prakarsh-dt — APPROVED** [review]

**Pull request merged**